mantidproject/mantid
# Add crystal density option to powder SampleDetails

**Description of work.**
This PR adds the option for the full crystallographic density to powder SampleDetails.
For converting between PDF types in polaris we need to use the "Full" or "Crystallographic" density of the sample which is different to the number dencity. In this PR the crystal density is stored in the SampleDetails and is obtained from that when needed.

**To test:**
Run the code below, there is a config file i can send over slack.
Try with dfferent crystal densities and compare the output pdf workspace, the output should scale with crystal density.
```
from mantid.simpleapi import *
import numpy as np
import matplotlib.pyplot as plt
from isis_powder import polaris, SampleDetails

config_file_path = r"C:/Users/wey38795/Documents/polaris-calculate-pdf/polaris_config_example.yaml"
polaris = polaris.Polaris(config_file=config_file_path, user_name="Test", mode="PDF")  

sample_details = SampleDetails(height=4.0, radius=0.2985, center=[0, 0, 0], shape='cylinder')
sample_details.set_material(chemical_formula='Si', number_density=0.123, crystal_density=0.246)
polaris.set_sample_details(sample=sample_details)

polaris.create_vanadium(first_cycle_run_no="98532",
                        multiple_scattering=False)
polaris.focus(run_number="98533", input_mode='Summed')

q_lim = [[2.5, 3, 4, 6, 7], [3.5, 5, 7, 11, 40]]
polaris.create_total_scattering_pdf(run_number='98533', 
                                    merge_banks=True, 
                                    q_lims=q_lim, 
                                    pdf_type='g(r)', 
                                    )
```

Fixes #28418 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
